File: py/7.py
from math import sqrt

def prime(n):
    if n == 2:
        return True

    if n % 2 == 0:
        return False

    cands = range(3, int(sqrt(n)) + 1)
    return all(map(lambda m: n % m, cands))

count = 0

for i in xrange(2, 105000):
    if prime(i):
        count += 1
    if (count == 10001):
        print i
        break

File: clj/7.clj
(defn prime? [n]
  (cond
    (= 2 n) true
    (even? n) false
    :else (every? #(not= 0 (mod n %)) (range 3 (inc (Math/sqrt n)) 2))))

(prn (last (take 10001 (filter prime? (iterate inc 2)))))
